optimum/habana/transformers/generation/utils.py

```diff
@@ -296,6 +319,55 @@ def _get_stopping_criteria(
         criteria = self._merge_criteria_processor_list(criteria, stopping_criteria)
         return criteria
 
+    @torch.no_grad()
```

Comment:
Line 283:

if (
                generation_config.static_shapes
                and self.config.is_encoder_decoder
                and self.generation_config.generation_mode == GenerationMode.GREEDY_SEARCH
            ):
                criteria.append(StaticMaxLengthCriteria(generation_config.max_length))


Need to add condition for Beam as well